CMakeLists.txt

```diff
@@ -13,8 +13,12 @@
 cmake_minimum_required(VERSION 2.8.4 FATAL_ERROR)
 cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION})
 
+IF (NOT DEFINED KFS_DIR_PREFIX)
+    set (KFS_DIR_PREFIX ${CMAKE_SOURCE_DIR})
```

Comment:
CMake has an [external project](https://cmake.org/cmake/help/v3.0/module/ExternalProject.html) cabaility which we should use instead. It doesn't make sense that qfs changes its build system because of some downstream dependency. That creates unnecessary build dependencies. Thoughts?